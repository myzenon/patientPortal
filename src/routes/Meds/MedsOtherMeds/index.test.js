import React from 'react'
import { MedsOtherMeds } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: true
            },
            medication_000293159: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: true
            },
            medication_000493990: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: true
            },
            medication_000747126: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: true
            },
            medication_665820312: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: true
            },
        },
        match: {
            params: {
                medId: '665820312'
            }
        },
    }
    const wrapper = render(
        <MedsOtherMeds { ...props} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('clicks yes | go to Enter new medication page', done => {
    const props = {
        cmr: { ...mockObj.cmr },
        match: {
            params: {
                medId: '665820312'
            }
        },
        push: (url) => {
            expect(url).toBe('/meds-new-medication')
            done()
        }
    }
    const wrapper = mount(
        <MedsOtherMeds { ...props} />
    )
    wrapper.find('button').first().simulate('click')
})

test('clicks no | go to HEDIS Checkin page', done => {
    const props = {
        cmr: { ...mockObj.cmr },
        match: {
            params: {
                medId: '665820312'
            }
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                otherMedsCompleted: true
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                otherMedsCompleted: true
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/prepare-Chapter4')
            done()
        }
    }
    const wrapper = mount(
        <MedsOtherMeds { ...props} />
    )
    wrapper.find('button').at(1).simulate('click')
    done()
})

// test('Show previous other-med that user entered on "Enter New Medication" page.', (done) => {
//     const props = {
//         patient: {
//             ...mockObj.patient
//         },
//         cmr: {
//             ...mockObj.cmr,
//             otherMedication_1: {
//                 display: "aspirin 200mg",
//                 brandName: "aspirin",
//                 genericName: "aspirin",
//                 dosage: { "value": "200mg" },
//                 taken: "y",
//                 reasonCode: [
//                     { conditionName: "Other", description: "Headache" }
//                 ],
//                 reviewCompleted: " true"
//             },
//             otherMedsCompleted: true
//         },
//         match: {
//             params: {
//                 medId: '665820312'
//             }
//         },
//     }
//
//     const wrapper = mount(
//         <MedsOtherMeds { ...props } />
//     )
//     setTimeout(() => {
//         expect(toJson(wrapper.render())).toMatchSnapshot()
//         done()
//     }, 0)
// })

test('YES shows "Meds Dosing Regimen" page if other-med unfinished.', done => {
    const props = {
        patient: {
            ...mockObj.patient
        },
        cmr: {
            ...mockObj.cmr,
            otherMedication_1: {
                display: "aspirin 200mg",
                brandName: "aspirin",
                genericName: "aspirin",
                taken: "y",
            },
        },
        push: (url) => {
            expect(url).toBe('/meds-new-medication/otherMedication_1')
            done()
        },
        match: {
            params: {
                medId: '665820312'
            }
        },
    }

    const wrapper = mount(
        <MedsOtherMeds { ...props } />
    )
    wrapper.find('button').first().simulate('click')
})

test('YES shows "Meds Any Other Questions" page if other-med unfinished.', done => {
    const props = {
        patient: {
            ...mockObj.patient
        },
        cmr: {
            ...mockObj.cmr,
            otherMedication_1: {
                display: "aspirin 200mg",
                brandName: "aspirin",
                genericName: "aspirin",
                taken: "y",
                dosage: { "value": "200mg" },
                reasonCode: [
                    { conditionName: "Other", "description": "Headache" }
                ]
            },
        },
        push: (url) => {
            expect(url).toBe('/meds-any-other-questions/otherMedication_1')
            done()
        },
        match: {
            params: {
                medId: '665820312'
            }
        },
    }

    const wrapper = mount(
        <MedsOtherMeds { ...props } />
    )
    wrapper.find('button').first().simulate('click')
})
