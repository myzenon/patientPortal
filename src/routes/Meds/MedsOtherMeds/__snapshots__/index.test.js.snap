// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <p />
    <span
      class="questions"
    >
      Are there additional medications or herbal supplements you are taking?
    </span>
    <p />
    <br />
    <div
      class="buttonWrapper"
    >
      <div
        class="row"
      >
        <div
          class="text-left col"
        >
          <button
            class="button btn btn-secondary"
            type="button"
          >
            YES
          </button>
        </div>
        <div
          class="text-right col"
        >
          <button
            class="button btn btn-secondary"
            type="button"
          >
            NO
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepCompleted"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="stepCompleted"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumberActive"
          >
            2
          </div>
        </div>
        <div
          class="stepActive"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumberActive"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
